plugin/session_buffer.py

```diff
@@ -405,5 +423,112 @@ def _present_diagnostics_async(
         for sv in self.session_views:
             sv.present_diagnostics_async()
 
+    # --- textDocument/semanticTokens ----------------------------------------------------------------------------------
+
+    def do_semantic_tokens_async(self, view: sublime.View) -> None:
+        # time.sleep(0.2)  # for testing with clangd
+        if not userprefs().semantic_highlighting:
+            return
+        if not self.session.has_capability("semanticTokensProvider"):
+            return
+        if self.session.get_capability('semanticTokensProvider.legend.tokenTypes') is None:
+            return
+        if self.session.get_capability('semanticTokensProvider.legend.tokenModifiers') is None:
+            return
```

Comment:
According to the spec this should never happen because those properties are non-optional and must be arrays. So checking for `None` values shouldn't be needed.

Reply:
I think I added this to satisfy mypy / Pyright. Could add some `type: ignore` comments alternatively.

Reply:
Doesn't look like it's due to mypy.

Reply:
Actually I already have these type ignore comments where necessary ;)
I will remove the check here.